### ISSUE TITLE
Inserting an enum value into an integer column yields "Incorrect integer value"

### ISSUE THREAD

**Issue by helmutschneider** [mysql-net/MySqlConnector#965]
The following program works correctly on 1.2.1 but not 1.3.2. I'm not sure if it's an intended change or a regression. Could not find anything relevant in the changelog.

```csharp
using System;
using System.Threading.Tasks;
using MySqlConnector;

namespace App
{
    enum CarKind
    {
        Toyota,
    }

    sealed class Program
    {
        const string CarSchemaSql = @"
          CREATE TABLE `car` (
              `id` BIGINT PRIMARY KEY NOT NULL AUTO_INCREMENT,
              `kind` TINYINT UNSIGNED NOT NULL
          )";

        const string InsertCarSql = @"INSERT INTO `car`(`kind`) VALUES (?)";

        async static Task Main(string[] args)
        {
            var builder = new MySqlConnectionStringBuilder
            {
                Server = "127.0.0.1",
                UserID = "root",
                Password = "",
                Database = "my_db",
            };

            var conn = new MySqlConnection(builder.ConnectionString);

            await conn.OpenAsync();

            using var tblCmd = conn.CreateCommand();
            tblCmd.CommandText = CarSchemaSql;

            await tblCmd.PrepareAsync();
            await tblCmd.ExecuteNonQueryAsync();

            using var insertCmd = conn.CreateCommand();
            insertCmd.CommandText = InsertCarSql;
            await insertCmd.PrepareAsync();

            var param = insertCmd.CreateParameter();
            param.Value = CarKind.Toyota;
            insertCmd.Parameters.Add(param);

            // Fails on 1.3.2 with
            //   Unhandled exception. MySqlConnector.MySqlException (0x80004005): Incorrect integer value: '' for column 'kind' at row 1
            await insertCmd.ExecuteNonQueryAsync();
        }
    }
}

```

```shell
Unhandled exception. MySqlConnector.MySqlException (0x80004005): Incorrect integer value: '' for column 'kind' at row 1
   at MySqlConnector.Core.ServerSession.ReceiveReplyAsyncAwaited(ValueTask`1 task) in /_/src/MySqlConnector/Core/ServerSession.cs:line 867
   at MySqlConnector.Core.ResultSet.ReadResultSetHeaderAsync(IOBehavior ioBehavior) in /_/src/MySqlConnector/Core/ResultSet.cs:line 50
   at MySqlConnector.MySqlDataReader.ActivateResultSet(CancellationToken cancellationToken) in /_/src/MySqlConnector/MySqlDataReader.cs:line 135
   at MySqlConnector.MySqlDataReader.CreateAsync(CommandListPosition commandListPosition, ICommandPayloadCreator payloadCreator, IDictionary`2 cachedProcedures, IMySqlCommand command, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/MySqlDataReader.cs:line 444
   at MySqlConnector.Core.CommandExecutor.ExecuteReaderAsync(IReadOnlyList`1 commands, ICommandPayloadCreator payloadCreator, CommandBehavior behavior, IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/Core/CommandExecutor.cs:line 60
   at MySqlConnector.MySqlCommand.ExecuteNonQueryAsync(IOBehavior ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/MySqlCommand.cs:line 266
```

**Comment by bgrainger** [comment]
> Could not find anything in relevant in the changelog.

I'm guessing the relevant change is that `IgnorePrepare` now defaults to `false`. Previously, calls to `Prepare` were actually ignored. If so, this may have been a bug in `Prepare` for a long time.

**Comment by bgrainger** [comment]
Also fails with MySQL Connector/NET 8.0.23 with `MySqlException: Incorrect arguments to mysqld_stmt_execute`.

**Comment by bgrainger** [comment]
Succeeds in MySqlConnector (not MySql.Data) by adding explicit type: `param.MySqlDbType = MySqlDbType.Byte;`.

**Comment by bgrainger** [comment]
Most likely, the following code needs to be updated to coerce parameter values to the column types returned by the server when preparing the statement: https://github.com/mysql-net/MySqlConnector/blob/b8334e34d43fc2462e0d3a2901d18ec0752f5dc2/src/MySqlConnector/Core/SingleCommandPayloadCreator.cs#L108-L124

**Comment by helmutschneider** [comment]
@bgrainger Interesting, thank you for debugging. If an implicit cast is necessary would expect it to be to the backing type of the enum and not the DB type. My example used `TINYINT` by coincidence. However, this is totally outside my realm and I understand if you need to support actual database enums too.

**Comment by bgrainger** [comment]
The real problem might actually be failure to correctly infer the `MySqlDbType` for any enum type.

**Comment by bgrainger** [comment]
Fixed in [1.3.3](https://github.com/mysql-net/MySqlConnector/releases/tag/1.3.3).
